server/app/routes/lecture.js

```diff
@@ -50,16 +60,43 @@ router.post('/', function(req, res, next) {
 
 router.put('/:lectureId', function(req, res, next) {
   var io = req.app.get('socketio');
+
   req.lecture.update(req.body)
   .then(function(updatedLecture) {
     if (req.body.startTime && !req.body.endTime) {
       io.emit('startLecture', updatedLecture);
+      res.status(201).json(updatedLecture);
     }
     else if (req.body.endTime) {
       io.emit('endLecture', updatedLecture);
+      res.status(201).json(updatedLecture);
+
+      Question.findAll({where: {lectureId: req.lecture.id}})
+      .then(function(response){
+        var subjectText = 'Feedback Questions: '+req.lecture.name;
+        var bodyText = response.map(function(question){
+          return question.text+": "+question.answer;
+        }).join("\n\n");
+        var mailOptions = {
+          from: 'fsa.feedback@gmail.com', 
+          to: req.lecture.lecturer,
+          subject: subjectText,
+          text: bodyText
+        };
+        transporter.sendMail(mailOptions, function(error, info){
+          if(error){
+            console.log("error: ", error);
+            res.json({yo: 'error'});
+          }
+          else{
+            console.log('Message sent: ' + info.response);
+          };
+        }); 
+      })
+      .catch(next);
     }
-    res.status(201).json(updatedLecture)
-  }).catch(next);
+  })
+  .catch(next);
```

Comment:
Hmm. It's too bad we've got nested `.then`s. Seems like it should be possible to refactor this into a single promise chain.